sql/plan/load_data.go

```diff
@@ -0,0 +1,379 @@
+	defaultFieldsEscapedByDelim    = "\\"
+	defaultLinesTerminatedByDelim  = "\n"
+	defaultLinesStartingByDelim    = ""
+)
+
+func (l *LoadData) Resolved() bool {
+	return l.Destination.Resolved()
+}
+
+func (l *LoadData) String() string {
+	pr := sql.NewTreePrinter()
+
+	_ = pr.WriteNode("LOAD DATA %s", l.File)
+	return pr.String()
+}
+
+func (l *LoadData) Schema() sql.Schema {
+	return l.Destination.Schema()
+}
+
+func (l *LoadData) Children() []sql.Node {
+	return []sql.Node{l.Destination}
+}
+
+// updateParsingConsts parses the LoadData object to update the 5 constants that are used for file parsing.
+func (l *LoadData) updateParsingConsts() error {
+	if l.Lines != nil {
+		ll := l.Lines
+		if ll.StartingBy != nil {
+			l.linesStartingByDelim = string(ll.StartingBy.Val)
+		}
+		if ll.TerminatedBy != nil {
+			l.linesTerminatedByDelim = string(ll.TerminatedBy.Val)
+		}
+	}
+
+	if l.Fields != nil {
+		lf := l.Fields
+
+		if lf.TerminatedBy != nil {
+			l.fieldsTerminatedByDelim = string(lf.TerminatedBy.Val)
+		}
+
+		if lf.EscapedBy != nil {
+			if len(string(lf.EscapedBy.Val)) > 1 {
+				return fmt.Errorf("error: LOAD DATA ESCAPED BY %s must be 1 character long", lf.EscapedBy)
+			}
+
+			l.fieldsEscapedByDelim = string(lf.EscapedBy.Val)
+		}
+
+		if lf.EnclosedBy != nil {
+			lfe := lf.EnclosedBy
+
+			if lfe.Optionally {
+				l.fieldsOptionallyDelim = true
+			}
+
+			if lfe.Delim != nil {
+				if len(string(lfe.Delim.Val)) > 1 {
+					return fmt.Errorf("error: LOAD DATA ENCLOSED BY must be 1 character long")
+				}
+
+				l.fieldsEnclosedByDelim = string(lfe.Delim.Val)
+			}
+		}
+	}
+
+	return nil
+}
+
+func (l *LoadData) RowIter(ctx *sql.Context, row sql.Row) (sql.RowIter, error) {
+	// TODO: Add the security variables for mysql
+
+	// Start the parsing by grabbing all the config variables.
+	err := l.updateParsingConsts()
+	if err != nil {
+		return nil, err
+	}
+
+	// TODO: Add tmpdir setting for mysql
+	var fileName = l.File
+	if l.Local {
+		fileName = Tmpfiledir + TmpfileName
+	}
+
+	file, err := os.Open(fileName)
+	if err != nil {
+		return nil, err
+	}
+
+	scanner := bufio.NewScanner(file)
+
+	// Set the split function for lines.
+	l.parseLines(scanner)
+
+	// Skip through the lines that need to be ignored.
+	for l.IgnoreNum > 0 && scanner.Scan() {
+		scanner.Text()
+		l.IgnoreNum--
+	}
+
+	return &loadDataIter{
+		scanner:                 scanner,
+		destination:             l.Destination,
+		ctx:                     ctx,
+		file:                    file,
+		local:                   l.Local,
+		fieldsTerminatedByDelim: l.fieldsTerminatedByDelim,
+		fieldsEnclosedByDelim:   l.fieldsEnclosedByDelim,
+		fieldsOptionallyDelim:   l.fieldsOptionallyDelim,
+		linesTerminatedByDelim:  l.linesTerminatedByDelim,
+		linesStartingByDelim:    l.linesStartingByDelim,
+	}, nil
+}
+
+// parseLines finds the delim that terminates each line and returns the overall line.
+func (l LoadData) parseLines(scanner *bufio.Scanner) {
```

Comment:
This is a bad name for this method.

Just make the splitFunc function top level, rename it `splitLines`, and call `scanner.Split(splitLines)` in RowIter